### PR TITLE
Pass LD_/DYLD_ env vars to debugserver (fixes #877)

### LINKED ISSUE #877: DYLD_ / LD_ environment variables will not be forwarded correctly on OS X
Hi,

Here's an interesting issue: https://youtrack.jetbrains.com/issue/GO-4001#comment=27-2237730

According to the information available here: https://stackoverflow.com/a/33589760/951414 it seems that OS X doesn't like forwarding any of the DYLD_ or LD_ to the applications anymore, for "security" reasons.

As such, when using the lldb backend on OS X, some applications will fail to launch. The IDE will forward the said environment variable when configured manually but I'm not sure if Delve itself needs to then do anything special to pick it up and then forward it itself to the lldb backend.

I'll try and work on a patch for this soon, but given my 0 experience with this stuff, any pointers / solutions would be welcomed.

(and yes, I know that the users can disable the "security" feature but I'm not sure what effects that will have)

### PR DESCRIPTION
**Pull request by dlsniper**
Unfortunately this is something I can't verify on my machine so I'll let the CI / users to do it for me.

### CODE REVIEW

**Comment by aarzilli** [comment]
Can you add a test for this? You could make a fixture that does:

```
dyldenv := os.GetEnv("DYLD_LIBRARY_PATH")
runtime.Breakpoint()
```

then in a test (in proc_test) set DYLD_LIBRARY_PATH to something and when the target process hits the breakpoint eval dyldenv and check that the value passed through.

**Comment by dlsniper** [comment]
Will do. Sorry for being lazy.

**Comment by dlsniper** [comment]
@aarzilli I've added a test. Thank you.

File: pkg/proc/proc_test.go [hunk 1]
```diff
@@ -2934,3 +2934,21 @@ func TestRecursiveNext(t *testing.T) {
 		}
 	})
 }
+
+func TestIssue877(t *testing.T) {
+	if runtime.GOOS != "darwin" && testBackend == "lldb" {
```
Comment (aarzilli) [inline pkg/proc/proc_test.go]: Can you add a comment describing what the test does "Tests that environment variables starting with DYLD_" are passed to the target on macOS". Thank you.

**Comment by dlsniper** [comment]
Done

**Pull request merged**